honeybee_radiance/lib/modifiers.py

```diff
@@ -4,68 +4,187 @@
 transmittance values in your model. There is no guarantee that these values exactly
 match what you are trying to model.
 """
+from ..modifier.material import Plastic, Glass, Glow
+from ._loadmodifiers import _rad_modifiers
 
-from honeybee_radiance.modifier.material import Plastic, Glass, Glow
+
+# establish variables for the default modifiers used across the library
+# and auto-generate modifiers if they were not loaded from default.mat
 
 # generic opaque modifiers - visible and solar
-generic_floor = Plastic.from_single_reflectance('generic_floor_0.20', 0.2)
-generic_wall = Plastic.from_single_reflectance('generic_wall_0.50', 0.5)
-generic_ceiling = Plastic.from_single_reflectance('generic_ceiling_0.80', 0.8)
-generic_door = Plastic.from_single_reflectance('generic_opaque_door_0.50', 0.5)
+try:
+    generic_floor = _rad_modifiers['generic_floor_0.20']
+except KeyError:
+    generic_floor = Plastic.from_single_reflectance('generic_floor_0.20', 0.2)
+    generic_floor.lock()
+    _rad_modifiers['generic_floor_0.20'] = generic_floor
```

Comment:
I'm a bit concerned about this approach to be a problem. The reason being there are two places that the default material might get assigned. What will be the case that default key is not available and we thing it should be considered ok? I think if the default key is not available we should throw an exception and ask for a fix.

Reply:
For all cases where the modifiers aren't critical to the functioning of the rest of the application, I agree that your approach makes sense. But people won't be able to import `honeybee_radiance` if one of these modifiers is missing since ModifierSets won't have the default modifiers they need and, by extension, the FaceProperties won't be importable. So there aren't any cases where it should be considered ok to have one of these particular modifiers missing.

Also, I know that there will be several common cases where we won't have the necessary .rad files to load the modifiers if we want to separate them into their own `honeybee_standards` package that we can install in a central user library that doesn't get overwritten with each update.  I guess we could just duplicate these .rad files between the new `honeybee_standards` repo and `honeybee_radiance` so that there's always a backup copy to load from if we don't find the user library. So maybe the choice is between this type of rad-file duplication or the one we currently have here where we catch the missing modifiers and create them in Python if we don't find them. Just knowing how critical it is to have these modifiers, I would rather do the latter. What do you think?

Reply:
> Also, I know that there will be several common cases where we won't have the necessary .rad files to load the modifiers if we want to separate them into their own honeybee_standards package that we can install in a central user library that doesn't get overwritten with each update.

I'm not sure if I fully understand why this should cause such a case when we won't have the modifiers.

I think this needs a longer conversation to figure out the best solution. What you have implemented right now suffice for this PR. Let's revisit this after we talk if necessary.